docs(inputs): revamp events articles (#1912)
telerik/blazor-docs: components/numerictextbox/events.md

This article explains the events available in the Telerik NumericTextbox for Blazor:

* [OnBlur](#onblur)
* [OnChange](#onchange)
* [ValueChanged](#valuechanged)

## OnBlur

The `OnBlur` event fires when the component loses focus.

>caption Handle the OnBlur event

````CSHTML
@* You do not have to use OnChange to react to loss of focus *@

@result

<TelerikNumericTextBox @bind-Value="@TheValue"
                       OnBlur="@OnBlurHandler">
</TelerikNumericTextBox>

@code {
    private string result = string.Empty;

    private decimal TheValue { get; set; } = 12.34m;

    private void OnBlurHandler()
    {
        result = $"BLUR fired, current value is {TheValue}.";
    }
}
````

## OnChange

The `OnChange` event represents a user action - confirmation of the current value. It fires when the user presses `Enter` in the input, or when the input loses focus.

> The NumericTextBox is a generic component, so you must provide either a `Value`, or a type to the `T` parameter of the component.

Reviewer: ```suggestion
The NumericTextBox is a generic component, so you must either provide a `Value`, or a type to the `T` parameter of the component.
```

This article explains the events available in the Telerik NumericTextbox for Blazor:

* [OnBlur](#onblur)
* [OnChange](#onchange)
* [ValueChanged](#valuechanged)

## OnBlur

The `OnBlur` event fires when the component loses focus.

>caption Handle the OnBlur event

````CSHTML
@* You do not have to use OnChange to react to loss of focus *@

@result

<TelerikNumericTextBox @bind-Value="@NumericTextBoxValue"
                       OnBlur="@OnBlurHandler">
</TelerikNumericTextBox>

@code {
    private string result = string.Empty;

    private decimal NumericTextBoxValue { get; set; } = 12.34m;

    private void OnBlurHandler()
    {
        result = $"BLUR fired, current value is {NumericTextBoxValue}.";
    }
}
````

## OnChange

> The `OnChange` event represents a user action - confirmation of the current value. It fires when the user presses `Enter` in the input, or when the input loses focus.

Reviewer 2: ```suggestion
The `OnChange` event represents a user action—confirmation of the current value. It fires when the user presses `Enter` in the input or when the input loses focus.
```

The NumericTextBox is a generic component, so you must either provide a `Value`, or a type to the `T` parameter of the component.

>caption Handle OnChange and use two-way binding

````CSHTML
@result
<br />
model value: @NumericTextBoxValue
<br />

<TelerikNumericTextBox @bind-Value="@NumericTextBoxValue" 
                       OnChange="@MyOnChangeHandler">
</TelerikNumericTextBox>

@code {
    private string result = string.Empty;

    private double NumericTextBoxValue { get; set; } = 1.2345;

    private void MyOnChangeHandler(object userInput)
    {
        // the handler receives an object that you may need to cast to the type of the component
        // when a Value is provided, the type is taken from it
        result = string.Format("The user entered: {0}", (double)userInput);
    }
}
````

@[template](/_contentTemplates/common/general-info.md#event-callback-can-be-async)

>tip The `OnChange` event is a custom event and does not interfere with bindings, so you can use it together with models and forms.

## ValueChanged

> The `ValueChanged` event fires upon every valid change in the input (for example, keystroke). Using this event requires one-way binding for the `Value` parameter and manual update of the value in the handler. If the value is not updated, this will effective cancel the event.

Reviewer 2: ```suggestion
The `ValueChanged` event fires upon every valid change in the input (for example, keystroke). Using this event requires one-way binding for the `Value` parameter and manual updating of the value in the handler. If the value is not updated, this will effectively cancel the event.
```
